Mathlib/RingTheory/Valuation/ValExtension.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2024 Jiedong Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang, Bichang Lei
+-/
+import Mathlib.RingTheory.Valuation.Integers
+import Mathlib.RingTheory.Ideal.LocalRing
+
+/-!
+# Extension of Valuation
+
+In this file, we define the typeclass for valuation extensions and prove basic facts about the
+extension of valuations. Let `A` be an `R` algebra, equipped with valuations `vA` and `vR`
+respectively. Here, the extension of a valuation means that the pullback of valuation `vA` to `R`
+is equivalent to the valuation `vR` on `R`. We only require equivalence, not equality, of
+valuations here.
+
+Note that we do not require the ring map from `R` to `A` to be injective. This holds automatically
+when `R` is a division ring and `A` is nontrivial.
+
+A motivation for choosing the more flexible `Valuation.Equiv` rather than strict equality here is
+to allow for possible normalization. As an example, consider a finite extension `K` of `ℚ_[p]`,
+which is a discretly valued field. We may choose the valuation on `K` from:
+
+1. the valuation where the uniformizer is mapped to one (more precisely, `-1` in `ℤₘ₀`) or
+
+2. the valuation where `p` is mapped to one.
+
+For the algebraic closure of `ℚ_[p]`, if we choose the valuation of `p` to be one, then the
+restriction of this valuation to `K` equals the second valuation, but is only equivalent to the
+first valuation. The flexibility of equivalence here allows us to develop theory for both cases
+without first determining the normalizations once and for all.
+
+## Main Definition
+
+* `IsValExtension vR vA` : The valuation `vA` on `A` is an extension of the valuation `vR` on `R`.
+
+## References
+
+* [Bourbaki, Nicolas. *Commutative algebra*] Chapter VI §3, Valuations.
+* <https://en.wikipedia.org/wiki/Valuation_(algebra)#Extension_of_valuations>
+
+## Tags
+Valuation, Extension of Valuations
+
+-/
+open Valuation
+
+variable {R A ΓR ΓA : Type*} [CommRing R] [Ring A]
+    [LinearOrderedCommMonoidWithZero ΓR] [LinearOrderedCommMonoidWithZero ΓA] [Algebra R A]
+    (vR : Valuation R ΓR) (vA : Valuation A ΓA)
+
+/--
+The class `IsValExtension R A` states that the valuation of `A` is an extension of the valuation
+on `R`. More precisely, the valuation on `R` is equivlent to the comap of the valuation on `A`.
+-/
+class IsValExtension : Prop where
+  /-- The valuation on `R` is equivlent to the comap of the valuation on `A` -/
+  val_isEquiv_comap : vR.IsEquiv <| vA.comap (algebraMap R A)
+
+namespace IsValExtension
+
+section algebraMap
+
+variable [IsValExtension vR vA]
+
+-- @[simp] does not work because `vR` cannot be inferred from `R`.
```

Comment:
Because of issues like this, I think it would be worthwhile to experiment with making `vR` an `outParam`.